test/hilbert/test_hilbert.py

```diff
@@ -325,6 +326,28 @@ def test_hilbert_index_discrete(hi: DiscreteHilbert):
         op.to_sparse()
 
 
+@partial(jax.jit, static_argnums=0)
+def _states_to_local_indices_jit(hilb, x):
+    return hilb.states_to_local_indices(x)
+
+
+@pytest.mark.parametrize("hi", discrete_hilbert_params)
+def test_states_to_local_indices(hi):
+    if isinstance(hi, nkx.hilbert.SpinOrbitalFermions):
+        pytest.xfail()
```

Comment:
I'll leave it to @jwnys to implement this for SpinOrbitalFermions.

Reply:
@PhilipVinc , why do you say "it cannot easily be extended for fermonic stuff."? Fermions are created by TensorHilbert of Fock space, both of which you have a "states_to_local_indices" implemented for..